### ISSUE TITLE
NATIVE_CHAR constant missing from h5py.h5t?

### ISSUE THREAD

**Issue by ZedThree** [h5py/h5py#632]
One way to store a string in HDF5 is as a 1D dataset of type H5T_NATIVE_CHAR (see [section 6.5.1](https://www.hdfgroup.org/HDF5/doc/UG/11_Datatypes.html) of the HDF5 docs). I guess the corresponding type in h5py should be `h5py.h5t.NATIVE_CHAR`, but this constant doesn't exist, so it doesn't seem possible to store strings like this with h5py, even with the low-level API.

Is this intentionally missing/unsupported?

**Comment by andrewcollette** [comment]
I don't think this was intentional, but H5T_NATIVE_CHAR is presumably equivalent to a UINT8, and would not show up in NumPy as a string type.  If you want an array of single text characters, try using length-1 fixed-width strings.  These will show up in NumPy as dtype "S1".
